Lib/test/libregrtest/utils.py

```diff
@@ -54,3 +56,102 @@ def printlist(x, width=70, indent=4, file=None):
     print(textwrap.fill(' '.join(str(elt) for elt in sorted(x)), width,
                         initial_indent=blanks, subsequent_indent=blanks),
           file=file)
+
+BUFSIZE = 8192
+LOAD_FACTOR_1 = 0.9200444146293232478931553241
+SAMPLING_INTERVAL = 5
+COUNTER_NAME = r'\System\Processor Queue Length'
+
+"""
+This class asynchronously interacts with the `typeperf` command to read
+the system load on Windows. Mulitprocessing and threads can't be used
+here because they interfere with the test suite's cases for those
+modules.
+"""
+class WindowsLoadTracker():
+    def __init__(self):
+        self.load = 0.0
+        self.start()
+
+    def start(self):
+        import _winapi
+        import msvcrt
+        import uuid
+
+        # Create a named pipe which allows for asynchronous IO in Windows
+        pipe_name =  r'\\.\pipe\typeperf_output_' + str(uuid.uuid4())
+
+        open_mode =  _winapi.PIPE_ACCESS_INBOUND
+        open_mode |= _winapi.FILE_FLAG_FIRST_PIPE_INSTANCE
+        open_mode |= _winapi.FILE_FLAG_OVERLAPPED
+
+        # This is the read end of the pipe, where we will be grabbing output
+        self.pipe = _winapi.CreateNamedPipe(
+            pipe_name, open_mode, _winapi.PIPE_WAIT,
+            1, BUFSIZE, BUFSIZE, _winapi.NMPWAIT_WAIT_FOREVER, _winapi.NULL
+        )
+        # The write end of the pipe which is passed to the created process
+        pipe_write_end = _winapi.CreateFile(
+            pipe_name, _winapi.GENERIC_WRITE, 0, _winapi.NULL,
+            _winapi.OPEN_EXISTING, 0x80000000, _winapi.NULL
+        )
+        # Open up the handle as a python file object so we can pass it to
+        # subprocess
+        command_stdout = msvcrt.open_osfhandle(pipe_write_end, 0)
+
+        # Connect to the read end of the pipe in overlap/async mode
+        overlap = _winapi.ConnectNamedPipe(self.pipe, overlapped=True)
+        overlap.GetOverlappedResult(True)
+
+        # Set up a job group so our child dies if we crash
+        limit_info = types.SimpleNamespace()
+        limit_info.LimitFlags = _winapi.JOB_OBJECT_LIMIT_KILL_ON_JOB_CLOSE
+        job_group = _winapi.CreateJobObject(limit_info)
+
+        # Spawn off the load monitor
+        command = ['typeperf', COUNTER_NAME, '-si', str(SAMPLING_INTERVAL)]
+        self.p = subprocess.Popen(command, stdout=command_stdout)
```

Comment:
Python still supports Windows 7, which only allows a process to be assigned to one job at a time. If the OS version is prior to NT 6.2 (i.e. `sys.getwindowsversion() < (6, 2)`), use the creation flag 
`CREATE_BREAKAWAY_FROM_JOB` (0x01000000) to try to break the child process out of the current job.

Reply:
There's no reason to break the child out of the job hierarchy in Windows 8+, especially since jobs are used more frequently (e.g. to implement silos, such as for Windows containers). 

Also, trying to break away from the parent job complicates creating the process. I forgot to mention that `Popen` will fail with a `PermissionError` if the child isn't allowed to break away. This can be handled by retrying without `CREATE_BREAKAWAY_FROM_JOB`. In this case you could skip calling `AssignProcessToJobObject`. For example (untested):

    CREATE_BREAKAWAY_FROM_JOB = 0x01000000

    cflags = 0 if sys.getwindowsversion() >= (6, 2) else CREATE_BREAKAWAY_FROM_JOB
    assign_to_job = True

    try:
        self.p = subprocess.Popen(command, stdout=command_stdout,
                    creationflags=cflags)
    except PermissionError:
        if cflags == 0:
            raise
        self.p = subprocess.Popen(command, stdout=command_stdout)
        assign_to_job = False

    if assign_to_job:
        _winapi.AssignProcessToJobObject(job_group, self.p._handle)